inst/include/dust/dust.hpp

```diff
@@ -15,12 +15,52 @@
 #include <omp.h>
 #endif
 
+
+// TODO: move these into a utilities file
+// TODO: make safe to use with references
+//
+// NB: these functions expect pointer for dest and ref for src,
+// so if applying to a vector use .data().
+// If the vector is passed in this will still compile due to the template,
+// but as not passed by ref will update a copy not the vector itself
```

Comment:
Don't know if there's any template magic that would give a compiler error in this case?
(this was one difficulty I mentioned)

Reply:
yes, you can fix that with a static assert I think provided we can find a trait that suits the type. Will think about this

Reply:
Done in 9f2bfd1, you will probably like this